java/arcs/android/storage/service/DevToolsBindingContext.kt

```diff
@@ -0,0 +1,48 @@
+package arcs.android.storage.service
+
+import arcs.android.devtools.IDevToolsService
+import arcs.android.storage.decodeProxyMessage
+import arcs.core.storage.ProxyMessage
+import arcs.core.storage.StorageKey
+import arcs.core.storage.Store
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * A wrapper around the [BindingContext] to send updates to DevTools.
+ */
+class DevToolsBindingContext(
+    /**
+     * The [Store] this [BindingContext] provides bindings for, it may or may not be shared with
+     * other instances of [BindingContext].
+     */
+    private val store: Store<*, *, *>,
+    /** [CoroutineContext] on which to build one specific to this [BindingContext]. */
+    parentCoroutineContext: CoroutineContext,
+    /** Sink to use for recording statistics about accessing data. */
+    private val bindingContextStatisticsSink: BindingContextStatisticsSink,
+    private val devToolsService: IDevToolsService?,
+    /** Callback to trigger when a proxy message has been received and sent to the store. */
+    private val onProxyMessage: suspend (StorageKey, ProxyMessage<*, *, *>) -> Unit = { _, _ -> }
+) : IStorageService.Stub() {
+
+    val parentBinding =
+        BindingContext(store, parentCoroutineContext, bindingContextStatisticsSink, onProxyMessage)
+
+    override fun idle(timeoutMillis: Long, resultCallback: IResultCallback) =
+        parentBinding.idle(timeoutMillis, resultCallback)
+
+    @Suppress("UNCHECKED_CAST")
+    override fun registerCallback(callback: IStorageServiceCallback) =
+        parentBinding.registerCallback(callback)
+
+    @Suppress("UNCHECKED_CAST")
+    override fun sendProxyMessage(
```

Comment:
This wrapper allows you to forward the proxy messages to DevTools. That's nice, but we know that we want more, e.g: notification when data gets flushed to the driver.

Instead of doing this wrapping, I would prefer if we explored adding an interface that allows Stores and Drivers to notify of events and in the future also respond to requests, according to the protocol we define - let's call it DevToolsBridge for the sake of this comment (you can pick whatever name you want).

If we do that, then the task is not to wrap the BindingContext, the task is extending the Store and Drivers with a instances of the bridge object that represents a debugging protocol and adding appropriate logging (and in the future request handling) in the right places.

Then, depending on the state of DevTools connecting in the StorageService, the Stores we create should get an implementation of this debug-bridge object that forwards to a DevToolsService, or a no-op (or null) instance.

How does that sound?